_use_cases/encryption-at-rest.markdown

```diff
@@ -0,0 +1,105 @@
+1. Kafka Client ecosystem is polyglot - it is common to Kafka applications written in many languages (Java, Go, Rust...)
+   deployed within a _single organisation_.  Any solution needs to work across all the target languages.
+1. There are some encryption libraries for some languages that can be used with the Kafka Clients but these available
+   across all languages.
+
+Organisations could write their own encryption code but this is a burden to the application teams. The need for
+_interoperability_ between different language encryption implementations - this increases the overheads.  There is also
+the problem of key distribution to consider.  Some part of the system needs to push the correct encryption keys out to
+the applications and manage tasks such as key rotation.  Finally, cryptography is a specialist area and the consequences
+of a design flaw or bug are significant (confidentiality breach).
+
+Having the applications encrypt/decrypt data themselves, whilst technically feasible, is not really a tenable solution
+at the scale required for most enterprises.
+
+# Kroxylicious Topic Encryption
+
+The Kroxylicious Topic Encryption feature offers a solution to the problem.  The proxy takes the responsibility
+to encrypt and decrypt the messages. In this way, the Kafka Brokers never see the plain text content of
+the message thus ensuring confidentiality.  Encryption is introduced into the system without requiring changes to
+either applications or the Kafka Cluster.
+
+Within Kroxylicious, the job of encryption and decryption is delegated to a Kroxylicious Filter. The filter intercepts
+produce requests from producing applications and encrypts the content, before the produce request is forwarded to the
+brokers.  For consuming, the filter intercepts fetch responses and decrypts the contents before the
+fetch response is forwarded to the applications.
+
+As the solution is proxy-based, it works regardless of the language the applications is written in.
+
+The solution's foundations rest on industry-standard encryption techniques.
+
+* Envelope Encryption[^1] is employed to efficiently encrypt/decrypt records.
+* Integrates with Key Management Services for safe and secure storage of key encryption keys.
+  * Plugin for [HashiCorp Vault](https://www.hashicorp.com/)&#174; 
+  * Additional KMS implementations are planned.
+  * API available to plugin alternatives
+* Uses AES-GCM symmetric keys.
+* Supports key rotation (new version applied to newly produced records only).
+* Records encrypted using previous key-versions remain decryptable.    
+
+[^1]: [NIST SP.-800-57 part 1 revision 5](https://csrc.nist.gov/Projects/Key-Management/Key-Management-Guidelines)
+
+Deployment time configuration at allows the administrator to choose which
+encryption keys are to be used to encrypt the records of which topics. Uses cases where some topics
+are encrypted and some remain unencrypted are supported.
```

Comment:
Also make the point that using a proxy enables flexibility in deployment. For example, you could use

* a single proxy for all applications
* a proxy-per application
* a proxy-per pod

which you chose depends on your requirements.

Reply:
As I said above, there is also flexibility in where the things are deployed.

Reply:
I think this comment needs to be addressed in the overview.